### DISCUSSION TITLE
**500 Internal Server Error** When Trying to Create a File in a Repository with REST API.

### DISCUSSION THREAD

**Discussion by artistconstellations-db** [community/community#36730] [Apps, API and Webhooks] [answered]
I am trying to [create a file in a public repository](https://docs.github.com/en/rest/repos/contents#create-or-update-file-contents) using the REST API in javascript (using the axios HTTP library). I've verified that the user and repo name are correct and that the filename does not already exist. **The 500 HTTP code isn't defined in the documentation so I'm not sure what to do.**

Here's the javascript that I'm using to make the request (simplified of course):
```javascript
  let response = await axios({
    method: "PUT",
    url:
      `https://api.github.com/repos/USER/REPO/contents/FILENAME`,
    headers: {
      "Accept": "application/vnd.github+json",
      "Authorization": `Bearer ${token}`,
    },
    data: {
      message: "a vague commit message here",
      content,
    },
  });
```

And the response I get in the browser console:

![image](https://user-images.githubusercontent.com/115185576/196840413-a6bbcf9e-bbad-4a20-81a8-c536602ad355.png)

*The reason I am not using the `octokit` library is that it does not build with node for browser in the framework that I am using, its `fetch` is not universal.*

**Comment by ossanna16** [comment]
Hey there @artistconstellations-db and thank you for asking a good question! I'm glad you found our community 🙂

**Reply by artistconstellations-db** [reply]
Thank you, do you know the answer to this question or someone who may know it?

**Reply by aleksandrefimov164** [reply]
@

**Comment by artistconstellations-db** [comment, marked as answer]
No one here was very helpful, but if anyone is reading this in the future and needs the answer, I figured it out:

Like in the documentation, you need to make sure the `content` in the PUT body is **base64**. That means that a javascript object can be converted like this:

```javascript
let content = {
    userData: "THIS IS SOME ARBITRARY DATA",
};

let contentCorrected = atob(JSON.stringify(content));
```

**Reply by phucbm** [reply]
Isn't the [`atob()`](https://www.w3schools.com/jsref/met_win_atob.asp) used for decoding?
